jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -103,6 +146,111 @@ public String toString()
         }
     }
 
+    /**
+     * Visitor
+     *
+     * A FileVisitor for walking a subtree of paths. The Scanner uses
+     * this to examine the dirs and files it has been asked to scan.
+     */
+    public class Visitor implements FileVisitor<Path>
+    {
+        Map<String, TimeNSize> scanInfoMap;
+        
+        public Visitor(Map<String, TimeNSize> scanInfoMap)
+        {
+            this.scanInfoMap = scanInfoMap;
+        }
+        
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException
+        {
+            File f = dir.toFile();
+            
+            if (!f.exists())
+                return FileVisitResult.SKIP_SUBTREE;
+
+            if (_reportDirs)
+            {
+                boolean accepted = false;
+                IncludeExcludeSet<PathMatcher, Path> includesExcludes = _scannables.get(dir);
+                if (includesExcludes != null)
+                { 
+                    //accepted if not explicitly excluded and either is explicitly included or there are no explicit inclusions
+                    Boolean result = includesExcludes.isIncludedAndNotExcluded(dir);
```

Comment:
I think the logic here is just `includesExcludes.test(dir)`

Reply:
Done